migration/data-differ/data-differ.py

```diff
@@ -39,9 +39,16 @@ def compare_docs_deepdiff(doc1, doc2, output_file):
 
 
 ## Main compare document function
-def compare_document_data(srcCollection, tgtCollection, batch_size, output_file, src_count):
-    source_cursor = srcCollection.find().sort('_id').batch_size(batch_size)
-    total_docs = src_count
+def compare_document_data(srcCollection, tgtCollection, batch_size, output_file, src_count, sample_size_percent):
+    if sample_size_percent:
+        percentage_in_decimal = sample_size_percent / 100
+        docs_to_sample = int(percentage_in_decimal * src_count)
+        source_cursor = srcCollection.aggregate([ { "$sample": { "size": docs_to_sample } } ], batchSize=batch_size, maxTimeMS=500)
```

Comment:
I'm not entirely sure what a good value for `maxTimeMS=500` is, but I figured you probably want to think about it, if your running queries that last more than half a second on your prod database

Reply:
We should probably have this as a configurable argument, if is set too low the cursor will error out like this: _pymongo.errors.ExecutionTimeout: PlanExecutor error during aggregation :: caused by :: operation exceeded time limit, full error: {'ok': 0.0, 'errmsg': 'PlanExecutor error during aggregation :: caused by :: operation exceeded time limit', 'code': 50, 'codeName': 'MaxTimeMSExpired'}._ 
According to the [docs](https://www.mongodb.com/docs/manual/reference/method/cursor.maxTimeMS/), maxTimeMS is  a cumulative time limit in milliseconds for processing operations on a cursor, so is not per batch.

On the other hand, I understand your point of protecting the prod db, it's a good option to have. If you have the time to push the change to the script  it will be appreciate, else, I'll pick it up in my next sprint.